From 610f5ddb3c4ae3b08ddb1334bbdcbcf7799f091f Mon Sep 17 00:00:00 2001
From: Susant Sahani <145210+ssahani@users.noreply.github.com>
Date: Wed, 18 Oct 2017 14:42:59 +0530
Subject: [PATCH] networkd: Consider linkLocalAddressing state while
 configuring ndisc. (#7012)

If linkLocalAddressing is disabled in for the interface still then
we look for ndisc configured or not in link_check_ready.

Link local is used automatic address configuration and neighbor discovery protocol.
If link local is disabled we should not configure ndisc.

Fixes #2713, #6441, #5841.

(cherry picked from commit 702c979fefc6d3e69c5982b8469ec8a47792d62f)
---
 src/network/networkd-link.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/network/networkd-link.c b/src/network/networkd-link.c
index 6b591271a0..c581a49b4e 100644
--- a/src/network/networkd-link.c
+++ b/src/network/networkd-link.c
@@ -226,6 +226,9 @@ static bool link_ipv6_accept_ra_enabled(Link *link) {
         if (!link->network)
                 return false;
 
+        if (!link_ipv6ll_enabled(link))
+                return false;
+
         /* If unset use system default (enabled if local forwarding is disabled.
          * disabled if local forwarding is enabled).
          * If set, ignore or enforce RA independent of local forwarding state.
